services/brig/src/Brig/API.hs

```diff
@@ -954,17 +954,16 @@ sitemap o = do
 
 setProperty :: UserId ::: ConnId ::: PropertyKey ::: JsonRequest PropertyValue -> Handler Response
 setProperty (u ::: c ::: k ::: req) = do
-    unless (Text.compareLength (Ascii.toText (propertyKeyName k)) maxKeyLen <= EQ) $
+    maxKeyLen <- fromMaybe 256 <$> view (settings . propertyMaxKeyLen)
+    maxValueLen <- fromMaybe 512 <$> view (settings . propertyMaxValueLen)
+    unless (Text.compareLength (Ascii.toText (propertyKeyName k)) (fromIntegral maxKeyLen) <= EQ) $
```

Comment:
the `fromIntegral` shouldn't be necessary.  we should just store the types expected here in `Settings`.